Add Sample Documentation (#55)
LM-Development/aks-sample: Samples/PublicSamples/RecordingBot/docs/guides/policy.md

The UPN is often also the email address of the user, but it does not have to be.
The upn of a user can be found in the user overview of the [Microsoft Entra Admin Center](https://entra.microsoft.com).

To verify if the policy was successfully assigned, you can run:

``` powershell
Get-CsOnlineUser <User Principal Name> | ft sipaddress, tenantid, TeamsComplianceRecordingPolicy
```

If the policy has been assigned successfully the output should look similar to

``` text
SipAddress                    TenantId                      TeamsComplianceRecordingPolicy
----------                    --------                      ------------------------------
sip:                          00000000-                     <Recording Policy Name>
```

### Assign the Policy to a group

[Grant-CsTeamsComplianceRecordingPolicy](https://learn.microsoft.com/powershell/module/teams/grant-csteamscompliancerecordingpolicy)

``` powershell
Grant-CsTeamsComplianceRecordingPolicy
      -Group <Group Object Id>
      -PolicyName <Recording Policy Name>
```

This assigns the policy to all users of the group specified by the object id of the group.

> Groups can be security groups and Microsoft 365 groups,

Author: I looked up the name in the azure portal and the Entra Id portal, groups can be either "Microsoft 365" or "Security". So I would say I left it purposfully, but I don't know if microsoft left it purposfully 😂

- assigned an Application in your own Entra (Application Instance)
  - wich points to an application on an external Entra (Application Id that was assigned to Application Instance)
- to a Teams Compliance Policy (by using its name)

## Use the policy

To be able to use the Policy, you will need to assign the policy to Users or Groups.

> [!NOTE]  
> It may take a few minutes and logged in users need a new access token (logout and login again) before the recording policy takes effect.

### Assign the Policy to a tenant

[Grant-CsTeamsComplianceRecordingPolicy](https://learn.microsoft.com/en-us/powershell/module/teams/grant-csteamscompliancerecordingpolicy?view=teams-ps)

``` powershell
Grant-CsTeamsComplianceRecordingPolicy 
      -Global 
      -PolicyName <Recording Policy Name>
```

This assigns the policy to all users of your tenant.

### Assign the Policy to a user

[Grant-CsTeamsComplianceRecordingPolicy](https://learn.microsoft.com/en-us/powershell/module/teams/grant-csteamscompliancerecordingpolicy?view=teams-ps)

``` powershell
Grant-CsTeamsComplianceRecordingPolicy 
    -Identity <User Principal Name> 
    -PolicyName <Recording Policy Name>
```

> This assigns the policy to the user specified by its user principal name(upn), the user principal name is often also the email of the user, but it doesn't have to, the upn of a user can be found in the user overview of the [Microsoft Entra Admin Center](https://entra.microsoft.com).

Reviewer: ```suggestion
This assigns the policy to the user specified by its user principal name(UPN).
The UPN is often also the email address of the user, but it does not have to be.
The upn of a user can be found in the user overview of the [Microsoft Entra Admin Center](https://entra.microsoft.com).
```

To verify if the policy was successfully assigned, you can run:

``` powershell
Get-CsOnlineUser <User Principal Name> | ft sipaddress, tenantid, TeamsComplianceRecordingPolicy
```

If the policy has been assigned successfully the output should look similar to

``` text
SipAddress                    TenantId                      TeamsComplianceRecordingPolicy
----------                    --------                      ------------------------------
sip:                          00000000-                     <Recording Policy Name>
```

### Assign the Policy to a group

[Grant-CsTeamsComplianceRecordingPolicy](https://learn.microsoft.com/en-us/powershell/module/teams/grant-csteamscompliancerecordingpolicy?view=teams-ps)

``` powershell
Grant-CsTeamsComplianceRecordingPolicy
      -Group <Group Object Id>
      -PolicyName <Recording Policy Name>
```

> This assigns the policy to all users of the group specified by the object id of the group. Groups can be security groups and Microsoft 365 groups, the object id of a group can be found in the group overview of the [Microsft Entra Admin Center](https://entra.microsoft.com)

Reviewer: ```suggestion
This assigns the policy to all users of the group specified by the object id of the group.
Groups can be security groups and Microsoft 365 groups,
the object id of a group can be found in the group overview of the [Microsft Entra Admin Center](https://entra.microsoft.com)
```

## Remove Recording Policy Assignment

> Removing a Recording Policy Assignment is very similar to assigning a recording Policy instead of specifying the Policy Name `$null` should be used for the Policy Name parameter.

Reviewer: ```suggestion
Removing a Recording Policy Assignment is very similar to assigning a recording Policy.
Passing `$null` as the parameter `PolicyName` will remove a Recording Policy.
```
